client.go

```diff
@@ -363,3 +394,113 @@ func JoinQueue[ResponseType proto.Message](
 	}
 	return sub, nil
 }
+
+func OpenStream[SendType, RecvType proto.Message](
+	ctx context.Context,
+	c *RPCClient,
+	rpc string,
+	topic string,
+	opts ...RequestOption,
+) (stream ClientStream[SendType, RecvType], err error) {
+
+	o := getRequestOpts(c.clientOpts, opts...)
```

Comment:
it seems like we might want a version of request options specific to streaming because the request/response model doesn't really apply here. it might have made sense to use message/ack but this would require exposing the internal protobufs so it won't work.

it might make sense to implement something like grpc client interceptors since the way the hooks are implemented currently wouldn't support the two use cases we planned for this - retries and adaptive timeouts based on observed rtt.